test/wdio.conf.js

```diff
@@ -31,6 +31,89 @@ async function capabilityDefaults() {
   };
 }
 
+class GitHubStatus {
+  /**
+   * @param {Object} p
+   * @param {string} p.sha
+   * @param {string} p.repo
+   * @param {string} p.token
+   * @param {string} p.context
+   */
+  constructor({sha, repo, token, context}) {
+    this.sha = sha;
+    this.repo = repo;
+    this.token = token;
+    this.context = context;
+  }
+
+  setTarget(priority, target) {
+    if (this.targetPriority > priority) {
+      return;
+    }
+
+    this.targetPriority = priority;
+    this.target = target;
+
+    if (this.lastStatus) {
+      this._send(this.lastStatus);
+    } else {
+      this.update({
+        state: 'pending'
+      })
+    }
+  }
+
+  update(status) {
+    this.lastStatus = status;
+    this._send(status);
+  }
+
+  _send(status) {
+    status.context = this.context;
+    status.target_url = this.target;
+    if (status.description && status.description.length > 140) {
+      status.description = status.description.substr(0, 140);
+    }
+
+    // Discard.
+    if (!this.sha || !this.repo || !this.token) {
+      this.token = '<masked>';
+      console.log('GitHub (status disabed)', this, status);
+      return;
+    }
+
+    console.log('GitHub status', status);
+
+    const api = this.repo.replace(/\/\/github.com\//, '//api.github.com/repos/');
+    const apiStatus = api + '/statuses/' + encodeURIComponent(process.env.COMMIT_SHA);
+    fetch(apiStatus, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/vnd.github+json',
+        'Authorization': 'Bearer ' + this.token,
+        'X-GitHub-Api-Version': '2022-11-28',
+      },
+      body: JSON.stringify(status),
+    })
+    .then(async res => {
+      if (!res.ok) {
+        throw new Error(res.url + ' returned status code ' + res.statusCode + ' and body: ' + (await res.text()));
+      }
+    })
```

Comment:
(It can't be just me who finds this sort of thing to be a complete mindfuck. The response returns a 2xx, and... _then_ you await...)

Reply:
That's right. We can process the headers immediately then decide whether we want to wait for the body.